### DISCUSSION TITLE
Crypt and  Decrypt with private key

### DISCUSSION THREAD

**Discussion by CryptoPowa** [MetaMask/snaps#880] [Ideas]
Add an API method to call to crypt some data and another to decrypt the data with the wallet private key stored in Metamask without getting it.

**Comment by ritave** [comment]
Hello @CryptoPowa, there's a lot of different ways such encryption could happen, each one with different trade-offs.

Do you have a specific one in mind?

We have a [Snaps Improvement Proposals](https://github.com/MetaMask/SIPs) programme, if you already have a specific API and implementation that you need - we'd appreciate you spending the time to write it down so that we can implement it.
